questions.ts

```diff
@@ -625,182 +642,229 @@ export const askQuestions = async (
       console.log("File or folder not found");
       process.exit(1);
     } else {
-      program.dataSource = {
-        type: fs.lstatSync(program.files).isDirectory() ? "folder" : "file",
-        config: {
-          paths: program.files.split(","),
+      program.dataSources = [
+        {
+          type: fs.lstatSync(program.files).isDirectory() ? "folder" : "file",
+          config: {
+            paths: program.files.split(","),
+          },
         },
-      };
+      ];
     }
   }
 
   if (!program.engine) {
     if (ciInfo.isCI) {
       program.engine = getPrefOrDefault("engine");
+      program.dataSources = getPrefOrDefault("dataSources");
     } else {
-      const { dataSource } = await prompts(
-        {
-          type: "select",
-          name: "dataSource",
-          message: "Which data source would you like to use?",
-          choices: getDataSourceChoices(program.framework),
-          initial: 1,
-        },
-        handlers,
-      );
-      // Initialize with default config
-      program.dataSource = getPrefOrDefault("dataSource");
-      if (program.dataSource) {
-        switch (dataSource) {
-          case "simple":
-            program.engine = "simple";
-            program.dataSource = { type: "none", config: {} };
-            break;
-          case "exampleFile":
-            program.engine = "context";
-            // Treat example as a folder data source with no config
-            program.dataSource = { type: "folder", config: {} };
-            break;
-          case "localFile":
-            program.engine = "context";
-            program.dataSource = {
-              type: "file",
-              config: {
-                paths: await selectLocalContextData("file"),
+      program.dataSources = [];
+      const numberDataSources = program.framework === "fastapi" ? 2 : 1;
+      for (let i = 0; i < numberDataSources; i++) {
+        const { selectedSource } = await prompts(
+          {
+            type: "select",
+            name: "selectedSource",
+            message:
+              i === 0
+                ? "Which data source would you like to use?"
+                : "Would you like to add another data source?",
+            choices: getDataSourceChoices(
+              program.framework,
+              program.dataSources,
+            ),
+            initial: 0,
+          },
+          handlers,
+        );
+
+        // Asking for data source config
+        // Select None data source, No need to config and asking for another data source
+        if (selectedSource === "none") {
+          if (selectedSource.length === 0) {
```

Comment:
is this correct? once `selectedSource` is a string and once an array

Reply:
Thanks, great catch! It should `program.dataSources.length` but i updated the code to make it easier to read.